Dockerfile

```diff
@@ -14,29 +14,55 @@ LABEL maintainer=development@kopano.io \
     org.label-schema.version=$ONE_VERSION \
     org.label-schema.schema-version="1.0"
 
+EXPOSE 9080 9443
+
+VOLUME ["/etc/kopano/", "/var/lib/mysql/", "/var/lib/kopano/"]
+
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 RUN apt-get update && \
-    apt-get install --no-install-recommends -y curl gnupg2 software-properties-common && \
+    apt-get install --no-install-recommends -y curl gnupg2 pwgen software-properties-common && \
     curl -fsSL $KOPANO_ONE_REPOSITORY_URL/$ONE_VERSION/gpg | apt-key add - && \
     add-apt-repository "deb $KOPANO_ONE_REPOSITORY_URL/$ONE_VERSION $(lsb_release -cs) supported" && \
+    apt-get upgrade -y -o Dpkg::Options::="--force-confold" && \
     rm -rf /var/cache/apt /var/lib/apt/lists/*
 
 RUN apt-get update && \
-    apt-get install --no-install-recommends -y kopano-one-$ONE_VERSION-groupware-packages && \
-    rm -rf /var/cache/apt /var/lib/apt/lists/*
+    apt-get install --no-install-recommends -y kopano-one-$ONE_VERSION kopano-smtpstd kopano-kidmd && \
+    rm -rf /var/cache/apt /var/lib/apt/lists/* && \
+# purge and re-create /var/lib/mysql with appropriate ownership
+	rm -rf /var/lib/mysql&& \
+	mkdir -p /var/lib/mysql&& \
+	chown -R mysql:mysql /var/lib/mysql&& \
+# comment out a few problematic configuration values
+	find /etc/mysql/ -name '*.cnf' -print0 \
+		| xargs -0 grep -lZE '^(bind-address|log|user\s)' \
+		| xargs -rt -0 sed -Ei 's/^(bind-address|log|user\s)/#&/'&& \
+# don't reverse lookup hostnames, they are usually another container
+	echo '[mysqld]\nskip-host-cache\nskip-name-resolve' > /etc/mysql/conf.d/docker.cnf
 
-ARG VCS_REF
-LABEL org.label-schema.vcs-ref=$VCS_REF
+# retrieve mariadb entrypoint
+RUN curl -o /usr/local/bin/mariadb-entrypoint.sh https://raw.githubusercontent.com/MariaDB/mariadb-docker/master/docker-entrypoint.sh && \
```

Comment:
This link should point to a specific Git revision or to a specific tag, to  avoid surprises.